multigeiger/speaker.cpp

```diff
@@ -58,17 +59,18 @@ void IRAM_ATTR isr_audio() {
       playing_tick = true;
     }
   }
+  portEXIT_CRITICAL_ISR(&mux_audio);
+
```

Comment:
iirc, isr_{sequence,tick_sequence,audio_sequence} are the global isr variables protected by mux_audio.

the code below is accessing isr_sequence, so it should be in the protected block.

Reply:
My understanding was that only writing to the critical variables should be protected. Must also the reading be protected?

Reply:
a mutex means that the protected blocks are mutually exclusive.

if writing is an issue and needs a mutex (likely because it is not atomic somehow), i guess one also must not read while a write is in progress. and if the read should be mutually excluded, it also needs to be in a mutex block.

Reply:
Okay, will move the mutex back

Reply:
also, I guess the mutex related calls also have quite a bit of overhead, so just keeping the mutex for some simple C operations might be more efficient than releasing and re-acquiring it more frequently.

Reply:
Right, makes sense. Kept the mutex active for the `!isr_sequence` poll though.